### PR TITLE
Fix Documents.getMany.queryIncludeDocs[T] bug 

### LINKED ISSUE #19: Document.getMany.queryIncludeDocs[T] fails if there is a design doc
`Documents.getMany.queryIncludeDocs[T]` `upickle.Invalid$Data` if the database has a design document or if the database contains a type different than `T`.

### PR DESCRIPTION
**Pull request by ermyas**
Fixes #19

### CODE REVIEW

**Comment by ermyas** [comment]
Would be nice to have a compiler warning generated when `getByTypeIncludeDocs[T]` is used. Something to the effect of "This method uses temporary views to query by type, which is inefficient. Use `getByTypeIncludeDocs[T](permView:CouchView)` for better performance" 

but for now a warning log might suffice

File: build.sbt [hunk 1]
```diff
@@ -45,7 +45,6 @@ scalacOptions ++= Seq(
   "-language:implicitConversions",
   "-language:postfixOps",
   "-unchecked",
-  "-Xfatal-warnings",
```
Comment (beloglazov) [inline build.sbt:48]: Could you please add a todo here to put this option back once the deprecated method is removed?

File: src/main/scala/com/ibm/couchdb/TypeMapping.scala [hunk 2]
```diff
@@ -16,7 +16,11 @@
 
 package com.ibm.couchdb
 
-final class TypeMapping private(val types: Map[String, String])
+final class TypeMapping private(val types: Map[String, String]) {
```
Comment (beloglazov) [inline src/main/scala/com/ibm/couchdb/TypeMapping.scala]: Should we then make the types field private and use the forType method everywhere instead?

File: src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala [hunk 3]
```diff
@@ -95,22 +111,55 @@ case class GetManyDocumentsQueryBuilder(client: Client,
...
+  def queryByTypeIncludeDocs[K, V, D: R](typeFilterView: CouchView)
+                                        (implicit tag: ClassTag[D], kr: R[K], kw: W[K], vr: R[V]):
+  Task[CouchDocs[K, V, D]] = {
+    typeMappings.forType(tag.runtimeClass) match {
+      case Some(k) => queryByType[K, V, D](typeFilterView, k)
+      case None => Res.Error("not_found", s"type mapping not found").toTask[CouchDocs[K, V, D]]
+    }
+  }
+
+  def queryByTypeIncludeDocs[D: R](implicit tag: ClassTag[D]):
```
Comment (beloglazov) [inline src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala]: What do you think about calling this something like queryByTypeIncludeDocsWithTemporaryView? :) It's kind of long, but it makes the difference with the other queryByTypeIncludeDocs clearer.

File: src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala [hunk 4]
```diff
@@ -95,22 +111,55 @@ case class GetManyDocumentsQueryBuilder(client: Client,
...
+  Task[CouchDocs[(String, String), String, D]] = {
+    log.warn(
+              "Only use `queryByTypeIncludeDocs[D: R]` for development purposes. It uses " +
+              "temporary views to perform type based filters and is inefficient. " +
+              "Instead, create a permanent view for type based filtering and use the " +
+              "`queryByTypeIncludeDocs[D: R] (typeFilterView: CouchView) method.")
```
Comment (beloglazov) [inline src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala]: The K and V type parameters are missing in the comment.

File: src/test/scala/com/ibm/couchdb/spec/Fixtures.scala [hunk 5]
```diff
@@ -32,6 +34,9 @@ trait Fixtures {
   val _docPersonName = _couchDoc composeLens _personName
   val _docPersonAge  = _couchDoc composeLens _personAge
 
+  val fixProfessorX = FixXPerson("Charles Xavier", "Professor X", "Telepathy, Astral projection, Mind control")
+  val fixMagneto    = FixXPerson("Max Eisenhardt", "Magneto", "Magnetism manipulation")
```
Comment (beloglazov) [inline src/test/scala/com/ibm/couchdb/spec/Fixtures.scala:38]: :+1:

File: src/test/scala/com/ibm/couchdb/api/DocumentsSpec.scala [hunk 6]
```diff
@@ -132,6 +132,21 @@ class DocumentsSpec extends CouchDbSpecification {
       docs.getDocsData mustEqual Seq(fixAlice, fixAlice)
     }
 
+    "Get all documents by type and include the doc data" >> {
```
Comment (beloglazov) [inline src/test/scala/com/ibm/couchdb/api/DocumentsSpec.scala:135]: Could you please also add a test for the other queryByTypeIncludeDocs? It's not covered right now.

File: build.sbt (hunk 1 above)
Comment (ermyas) [inline build.sbt:48]: :+1:

File: src/main/scala/com/ibm/couchdb/TypeMapping.scala (hunk 2 above)
Comment (ermyas) [inline src/main/scala/com/ibm/couchdb/TypeMapping.scala]: yeap, makes sense

File: src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala (hunk 3 above)
Comment (ermyas) [inline src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala]: sounds a bit too long (40 chars) :-) ... might require a line on its own :-D ... but I take your point ... will change

**Comment by ermyas** [comment]
change requests actioned :-)

**Pull request merged**